### PR TITLE
core: check known relays before file reception, support user approval of unknown relays

### PR DESCRIPTION
**Pull request by spaced4ndy**

### CODE REVIEW

File: src/Simplex/Chat/Controller.hs [hunk 1]
```diff
@@ -1110,6 +1110,7 @@ data ChatErrorType
   | CEFileImageType {filePath :: FilePath}
   | CEFileImageSize {filePath :: FilePath}
   | CEFileNotReceived {fileId :: FileTransferId}
+  | CEFileAbortedNotApproved {fileId :: FileTransferId, unknownServers :: [XFTPServer]}
```
Comment (epoberezkin) [inline src/Simplex/Chat/Controller.hs:1113]:
```suggestion
  | CEFileNotApproved {fileId :: FileTransferId, unknownServers :: [XFTPServer]}
```

File: src/Simplex/Chat/Messages.hs [hunk 2]
```diff
@@ -529,6 +529,7 @@ data CIFileStatus (d :: MsgDirection) where
   CIFSRcvInvitation :: CIFileStatus 'MDRcv
   CIFSRcvAccepted :: CIFileStatus 'MDRcv
   CIFSRcvTransfer :: {rcvProgress :: Int64, rcvTotal :: Int64} -> CIFileStatus 'MDRcv
+  CIFSRcvAborted :: CIFileStatus 'MDRcv
```
Comment (epoberezkin) [inline src/Simplex/Chat/Messages.hs:532]:
```suggestion
  CIFSRcvNotApproved :: CIFileStatus 'MDRcv
```

Comment (epoberezkin) [reply src/Simplex/Chat/Messages.hs:532]: Aborted is unclear here, better to use the same term consistently

File: src/Simplex/Chat/Controller.hs (hunk 1 above)
Comment (epoberezkin) [reply src/Simplex/Chat/Controller.hs:1113]: Aborted is unnecessary here

File: src/Simplex/Chat/Messages.hs (hunk 2 above)
Comment (spaced4ndy) [reply src/Simplex/Chat/Messages.hs:532]: this does happen when it's aborted though, could be reused for other errors possibly

**Pull request merged**